### ISSUE TITLE
Wrong logic for role assignment with oauth_restrict_* strings

### ISSUE THREAD

**Issue by carlosms** [src-d/code-annotation#101]
In OAuth.GetUser: Right now if restrictRequesterAccess is empty, restrictAccess is not checked.

**Comment by dpordomingo** [comment]
should we close this issue?
or should we redefine the scope of this issue?

**Comment by bzz** [comment]
Up to @carlosms, but I would expect either to be updated \w the scope of the work on testing, like
 - [x] fix ...
 - [x] refactor, so it's mockable
 - [x] add tests \w mocks

or to be closed and a new issue created for further work.

**Comment by carlosms** [comment]
PR with tests is here: https://github.com/src-d/code-annotation/pull/108

Once that is approved, we can close this issue.

**Comment by bzz** [comment]
Seems like #108 is ready to be merged now

**Comment by carlosms** [comment]
Everything is in master now
